### DISCUSSION TITLE
Async Auto-index

### DISCUSSION THREAD

**Discussion by natankeddem** [zauberzeug/nicegui#2101] [Q&A] [answered]
You can do:
```py
@ui.page('/')
async def index(client: Client):
    ui.label('Hello World!')
    await client.connected()
```
This will let you render a routed page as an `async` function to build your private page. Is it possible to have an auto-indexing `async` function to build the page?

**Comment by rodja** [comment, marked as answer]
The auto-index page is build with one client for all connections. Therefore you won't be able to to await the clients connection. But you could add elements in an async function which is executed by timer or on_startup handler:

```py
from nicegui import ui, app

ui.label('auto-index page')

async def add_content():
    await asyncio.sleep(5)
    ui.label('some content')

app.on_startup(add_content)

ui.run()
```

**Reply by natankeddem** [reply]
Good thoughts to utilize those events, thank you.
